### ISSUE TITLE
TypeError: write() argument must be str, not bytes

### ISSUE THREAD

**Issue by luosainan96** [wifiphisher/wifiphisher#1263]
Before opening a new issue, please make sure that what you are about to report has not been reported already or is not included on the FAQ page. If not, use the following template for reporting.

Please keep in mind that our developers will prioritize your issue if you provide a detailed and extensive description.

**Version**: Use `git show | grep commit` command and paste here the output.

**Description**: It occurred "fout.write(rfke) TypeError: write() argument must be str, not bytes" when I ran 'wifiphisher' on ubuntu16.04 after installation in python3.5. I only have one it didin't work that just change the file from 'w' to 'wb+'. I installed wifiphisher from git. only one physical interface.

**Script output**: 
[*] Starting Wifiphisher 1.4GIT ( https://wifiphisher.org ) at 2020-02-08 13:46
[+] Timezone detected. Setting channel range to 1-13
[+] Selecting wfphshr-wlan0 interface for the deauthentication attack
[+] Selecting wlp6s0 interface for creating the rogue Access Point
[+] Changing wlp6s0 MAC addr (BSSID) to 00:00:00:1b:8c:a7
[+] Changing wlp6s0 MAC addr (BSSID) to 00:00:00:3b:36:65
Traceback (most recent call last):
  File "/usr/local/bin/wifiphisher", line 9, in <module>
    load_entry_point('wifiphisher==1.4', 'console_scripts', 'wifiphisher')()
  File "/usr/local/lib/python3.5/dist-packages/wifiphisher-1.4-py3.5.egg/wifiphisher/pywifiphisher.py", line 709, in run
    engine.start()
  File "/usr/local/lib/python3.5/dist-packages/wifiphisher-1.4-py3.5.egg/wifiphisher/pywifiphisher.py", line 486, in start
    self.network_manager.unblock_interface(ap_iface)
  File "/usr/local/lib/python3.5/dist-packages/wifiphisher-1.4-py3.5.egg/wifiphisher/common/interfaces.py", line 629, in unblock_interface
    pyw.unblock(card)
  File "/usr/local/lib/python3.5/dist-packages/pyric/pyw.py", line 628, in unblock
    rfkill.rfkill_unblock(idx)
  File "/usr/local/lib/python3.5/dist-packages/pyric/utils/rfkill.py", line 135, in rfkill_unblock
    fout.write(rfke)
TypeError: write() argument must be str, not bytes

==================iw list:=========================
Wiphy phy0
	max # scan SSIDs: 4
	max scan IEs length: 2257 bytes
	Retry short limit: 7
	Retry long limit: 4
	Coverage class: 0 (up to 0m)
	Device supports RSN-IBSS.
	Device supports AP-side u-APSD.
	Device supports T-DLS.
	Supported Ciphers:
		* WEP40 (00-0f-ac:1)
		* WEP104 (00-0f-ac:5)
		* TKIP (00-0f-ac:2)
		* CCMP (00-0f-ac:4)
		* 00-0f-ac:10
		* GCMP (00-0f-ac:8)
		* 00-0f-ac:9
		* CMAC (00-0f-ac:6)
		* 00-0f-ac:13
		* 00-0f-ac:11
		* 00-0f-ac:12
	Available Antennas: TX 0x1 RX 0x3
	Configured Antennas: TX 0x1 RX 0x3
	Supported interface modes:
		 * IBSS
		 * managed
		 * AP
		 * AP/VLAN
		 * WDS
		 * monitor
		 * mesh point
		 * P2P-client
		 * P2P-GO
		 * Unknown mode (11)
	Band 1:
		Capabilities: 0x116e
			HT20/HT40
			SM Power Save disabled
			RX HT20 SGI
			RX HT40 SGI
			RX STBC 1-stream
			Max AMSDU length: 3839 bytes
			DSSS/CCK HT40
		Maximum RX AMPDU length 65535 bytes (exponent: 0x003)
		Minimum RX AMPDU time spacing: 8 usec (0x06)
		HT TX/RX MCS rate indexes supported: 0-7
		Bitrates (non-HT):
			* 1.0 Mbps
			* 2.0 Mbps (short preamble supported)
			* 5.5 Mbps (short preamble supported)
			* 11.0 Mbps (short preamble supported)
			* 6.0 Mbps
			* 9.0 Mbps
			* 12.0 Mbps
			* 18.0 Mbps
			* 24.0 Mbps
			* 36.0 Mbps
			* 48.0 Mbps
			* 54.0 Mbps
		Frequencies:
			* 2412 MHz [1] (16.0 dBm)
			* 2417 MHz [2] (16.0 dBm)
			* 2422 MHz [3] (16.0 dBm)
			* 2427 MHz [4] (16.0 dBm)
			* 2432 MHz [5] (16.0 dBm)
			* 2437 MHz [6] (16.0 dBm)
			* 2442 MHz [7] (16.0 dBm)
			* 2447 MHz [8] (16.0 dBm)
			* 2452 MHz [9] (16.0 dBm)
			* 2457 MHz [10] (16.0 dBm)
			* 2462 MHz [11] (16.0 dBm)
			* 2467 MHz [12] (16.0 dBm)
			* 2472 MHz [13] (16.0 dBm)
			* 2484 MHz [14] (18.0 dBm) (no IR)
	Supported commands:
		 * new_interface
		 * set_interface
		 * new_key
		 * start_ap
		 * new_station
		 * new_mpath
		 * set_mesh_config
		 * set_bss
		 * authenticate
		 * associate
		 * deauthenticate
		 * disassociate
		 * join_ibss
		 * join_mesh
		 * remain_on_channel
		 * set_tx_bitrate_mask
		 * frame
		 * frame_wait_cancel
		 * set_wiphy_netns
		 * set_channel
		 * set_wds_peer
		 * tdls_mgmt
		 * tdls_oper
		 * probe_client
		 * set_noack_map
		 * register_beacons
		 * start_p2p_device
		 * set_mcast_rate
		 * channel_switch
		 * Unknown command (104)
		 * connect
		 * disconnect
	Supported TX frame types:
		 * IBSS: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * managed: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * AP: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * AP/VLAN: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * mesh point: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * P2P-client: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * P2P-GO: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * P2P-device: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
	Supported RX frame types:
		 * IBSS: 0x40 0xb0 0xc0 0xd0
		 * managed: 0x40 0xd0
		 * AP: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
		 * AP/VLAN: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
		 * mesh point: 0xb0 0xc0 0xd0
		 * P2P-client: 0x40 0xd0
		 * P2P-GO: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
		 * P2P-device: 0x40 0xd0
	software interface modes (can always be added):
		 * AP/VLAN
		 * monitor
	valid interface combinations:
		 * #{ managed } <= 2048, #{ AP, mesh point } <= 8, #{ P2P-client, P2P-GO } <= 1,
		   total <= 2048, #channels <= 1, STA/AP BI must match
		 * #{ WDS } <= 2048,
		   total <= 2048, #channels <= 1, STA/AP BI must match
	HT Capability overrides:
		 * MCS: ff ff ff ff ff ff ff ff ff ff
		 * maximum A-MSDU length
		 * supported channel width
		 * short GI for 40 MHz
		 * max A-MPDU length exponent
		 * min MPDU start spacing
	Device supports TX status socket option.
	Device supports HT-IBSS.
	Device supports SAE with AUTHENTICATE command
	Device supports low priority scan.
	Device supports scan flush.
	Device supports AP scan.
	Device supports per-vif TX power setting
	P2P GO supports CT window setting
	Driver supports a userspace MPM
	Device supports active monitor (which will ACK incoming frames)
	Driver/device bandwidth changes during BSS lifetime (AP/GO mode)

==============iwconfig========================
enp7s0    no wireless extensions.

lo        no wireless extensions.

wlp6s0    IEEE 802.11bgn  ESSID:"Redmi K20 Pro_83va\xE2\x80\x8B"  
          Mode:Managed  Frequency:2.432 GHz  Access Point: E0:DC:FF:51:83:B7   
          Bit Rate=65 Mb/s   Tx-Power=15 dBm   
          Retry short limit:7   RTS thr:off   Fragment thr:off
          Encryption key:off
          Power Management:off
          Link Quality=70/70  Signal level=-39 dBm  
          Rx invalid nwid:0  Rx invalid crypt:0  Rx invalid frag:0
          Tx excessive retries:5  Invalid misc:108   Missed beacon:0

wfphshr-wlan0  IEEE 802.11bgn  ESSID:off/any  
          Mode:Managed  Access Point: Not-Associated   Tx-Power=15 dBm   
          Retry short limit:7   RTS thr:off   Fragment thr:off
          Encryption key:off
          Power Management:off

**Comment by luosainan96** [comment]
The question can be solved by changing the  ''fout = open(dpath, 'w')'' to ''fout = open(dpath, 'wb')'' twice in the file:/usr/local/lib/python3.5/dist-packages/pyric/utils/rfkill.py

**Comment by sophron** [comment]
Thanks, we'll have a look.

**Comment by DorianUzun** [comment]
@luosainan96 the better solution is adding the py3 check at the open() call that is missing it:
`if _PY3_: rfke = rfke.decode('ascii')`

One of the of two instances in the rfkill.py file has this, the other does not.
Your solution will actually break the one that does. This patch I made fixes the issue for me with python3.5 (on yocto - morty):
```
--- a/pyric/utils/rfkill.py	Fri Feb 26 15:25:14 2021
+++ a/pyric/utils/rfkill.py	Fri Feb 26 15:24:46 2021
@@ -133,0 +134 @@ def rfkill_unblock(idx):
+        if _PY3_: rfke = rfke.decode('ascii')
```

**Comment by ThanatosXingYu** [comment]
> 

Thanks GOD! I sovle this problem that are the same as me successfully!!!!!

**Comment by sophron** [comment]
Everyone, thank you for your help. If you make a PR, I will merge this.

**Comment by spearium** [comment]
Guys help, i am new here and i dont know how to fix this error, so i would appreciate if any one can show me step by step
